src/snr_measurements.py

```diff
@@ -305,8 +306,163 @@ def check_xt(self):
+            elif 5 < self.spectrum_props.core_num  < 12 and self.engine_props['cores_per_link'] == 13:
+                loaded_data = np.load('MF-USB6014-MCF13-C5.npy', allow_pickle=True)
+                loaded_data_gsnr = np.load('GSNR-USB6014-MCF13-C5.npy', allow_pickle=True)
+            elif self.spectrum_props.core_num  == 12 and self.engine_props['cores_per_link'] == 13:
+                loaded_data = np.load('MF-USB6014-MCF13-C6.npy', allow_pickle=True)
+                loaded_data_gsnr = np.load('GSNR-USB6014-MCF13-C6.npy', allow_pickle=True)
+            elif self.spectrum_props.core_num >= 12 and self.engine_props['cores_per_link'] == 19:
+                loaded_data = np.load('MF-USB6014-MCF19-C6.npy', allow_pickle=True)
+                loaded_data_gsnr = np.load('GSNR-USB6014-MCF19-C6.npy', allow_pickle=True)
+            elif self.spectrum_props.core_num  in [0,2,4,6,8,10] and self.engine_props['cores_per_link'] == 19:
+                loaded_data = np.load('MF-USB6014-MCF19-C3.npy', allow_pickle=True)
+                loaded_data_gsnr = np.load('GSNR-USB6014-MCF19-C3.npy', allow_pickle=True)
+            elif self.spectrum_props.core_num  in [1,3,5,7,9,11] and self.engine_props['cores_per_link'] == 19:
+                loaded_data = np.load('MF-USB6014-MCF19-C4.npy', allow_pickle=True)
+                loaded_data_gsnr = np.load('GSNR-USB6014-MCF19-C4.npy', allow_pickle=True)
+        slot_index = 0
+        if self.spectrum_props.curr_band == 'l':
+            slot_index = self.spectrum_props.start_slot
+        elif self.spectrum_props.curr_band == 'c':
+            slot_index = self.engine_props['l_band'] + self.spectrum_props.start_slot
+        elif self.spectrum_props.curr_band == 's':
+            slot_index = (self.engine_props['l_band'] + 
+                          self.engine_props['c_band'] + 
+                          self.spectrum_props.start_slot)
+        else:
+            NotImplementedError(f"Unexpected band: {self.spectrum_props.curr_band}")
+        mod_format = loaded_data[self.route_props.connection_index[0]][slot_index][path_index]
+        SNR_val = loaded_data_gsnr[self.route_props.connection_index[0]][slot_index][path_index]
+        if mod_format == 0:
+            resp = False
+        elif mod_format_mapping[mod_format] == self.spectrum_props.modulation and BW_mapping[self.spectrum_props.modulation] >= int(self.sdn_props.bandwidth):
+            resp = True
+        else:
+            resp = False
+        return resp, SNR_val
+        raise NotImplementedError(f"Unexpected snr_type flag got: {self.engine_props['snr_type']}")
+
+
+    def check_snr_ext_slicing(self, path_index):
+        """
+        Checks the SNR on a single request using the external resources.
 
-    def handle_snr(self):
+        :return: Whether the SNR threshold can be met and SNR value.
+        :rtype: tuple
+        """
+        mod_format_mapping = {
+        6: "64-QAM",
```

Comment:
Same comments as above, changing this will help the organization when debugging along with the ability to scale.